grid-client/deployer/deployment_deployer.go

```diff
@@ -266,12 +291,10 @@ func (d *DeploymentDeployer) Validate(ctx context.Context, dl *workloads.Deploym
 	return dl.Validate()
 }
 
-func (d *DeploymentDeployer) assignNodesIPs(dl *workloads.Deployment) error {
+func (d *DeploymentDeployer) assignNodesIPs(dl *workloads.Deployment, usedHosts *[]byte) error {
```

Comment:
Why not return the array instead of using this 

```go
fn modeifyArray(input []byte) []byte {
  input = appen(input, 0)
  return input
}
```
It's both cleaner to read and implement